Docs/StyleGuide/Style-Guide.md

```diff
@@ -0,0 +1,280 @@
+    // to a string.
+    //
+    // this method name can be simplified to ToString()
+    // so that the method call is Integer.ToString(i)
+    // instead of Integer.IntegerToString(i).
+
+    // YES
+    method ToString(i: int) returns (s: string)
+        ...
+
+    // NO
+    method IntegerToString(i: int) returns (s: string)
+        ...
+}
+```
+
+## Code Layout
+
+### Braces
+Dafny by default opens a new brace in the same line.
+```dafny
+module M() {
+    ...
+    method Met() {
+        ...
+    }
+}
+```
+In case the method (or function, lemma, etc) signature is too long to fit in one line, or in case the signature has at
+least one specification clause, the new brace goes to the new line.
+
+```dafny
+module M() {
+    ...
+    method Met(i: int) returns (j: int)
+        requires i % 2 == 0
+        ensures j > 10
+    {
+        ...
+    }
+}
+```
+
+This applies to every scope: `module`, `class`, `predicate`, `if`, `while`, and more.
+
+### Imports
+Import the modules with `opened` to avoid typing a full module name each time a method is invoked.
+```dafny
+import opened StandardLibrary
+...
+```
+
+When there are more than two modules with the same method name, specify the name of the module during method invocation.
+
+```dafny
+import opened MyModule
+import opened YourModule
+...
+method MyMethod() {
+    MyModule.foo();
+    YourModule.foo();
+}
+```
+
+In case typing the full module name is cumbersome, import it with a shorthanded name.
+
+```dafny
+import M = MyModuleWithACumbersomeName
+import Y = YourModuleWithACumbersomeName
+...
+method MyMethod() {
+    M.foo();
+    Y.foo();
+}
+```
+
+Common imports, such as `StandardLibrary` and `Native`, should be grouped together, followed by custom module imports
+with a blank line in-between.
+
+```dafny
+import opened StandardLibrary
+import opened Native
+
+import opened Donut
+import Coffee
+```
+
+## Indentation and Line Breaks
+
+### Tabs or Spaces?
+Spaces are preferred over tabs. Tabs should only be used to remain consistent with the existing code with tabs.
+
+Use 4 spaces for each indentation.
+
+### Maximum Character Limit
+Although there is no strict requirement, it is generally recommended to have maximum of 120 characters in each line.
+
+### Functions, Methods, Predicates, and Lemmas
+Every Dafny method has the following signature.
+```dafny
+method {:<attributes>} MethodName(param1: Type, param2: Type) returns (ret: Type)
+    decreases param1
+    requires P()
+    modifies param2
+    ensures Q()
+```
+
+When possible, put MethodName and `returns` statement in the same line, as the keyword `returns` is distinct from other
+method specification clauses, such as `modifies`, `requires`, `ensures`, and `decreases`, which can be mixed in any
+order (although not recommended for readability), while `returns` should always be right after the parameter
+definitions. Each method specification clause should be in a separate line.
```

Comment:
I'd prefer this:

When possible, put `MethodName` and `returns` statement on the same line, as the keyword `returns` is distinct from other method specification clauses, such as `decreases`, `requires`, `modifies`,  and `ensures`, which should appear in this order. Each method specification clause should be on a separate line.

Reply:
Minor grammatical point:  put `MethodName` and *the* `returns` statement on the same line.